### PR TITLE
Remove hparams from test_ddp

### PR DESCRIPTION
**Pull request by hanlint**
This PR removes the usage of `yahp` in the DDP unit testing.

Coming along for the ride: some readability and stylistic fixes

### CODE REVIEW

File: tests/trainer/test_ddp.py [hunk 1]
```diff
@@ -189,20 +174,17 @@ def test_ddp(device: str, world_size: int, dummy_model_hparams: ModelHparams, de
                       deepspeed_config={} if deepspeed else None,
                       callbacks=[CheckBatch0(tmp_path)])
 
-    for evaluator in trainer.state.evaluators:
-        assert isinstance(evaluator.dataloader, DataSpec)
-        assert isinstance(evaluator.dataloader.dataloader, collections.abc.Sized)
     trainer.fit()
 
     expected_train_num_loads = max_epochs * train_batch_size * train_subset_num_batches
     #expected_val_num_loads = max_epochs * hparams.eval_batch_size * hparams.eval_subset_num_batches
```
Comment (nik-mosaic) [inline tests/trainer/test_ddp.py:180]: Remove?

File: tests/trainer/test_ddp.py [hunk 2]
```diff
@@ -189,20 +174,17 @@ def test_ddp(device: str, world_size: int, dummy_model_hparams: ModelHparams, de
...
     expected_val_num_loads = 0
-    for evaluator in trainer.state.evaluators:
+    for _ in trainer.state.evaluators:
         expected_val_num_loads += max_epochs * eval_batch_size * eval_subset_num_batches
 
     # adding hparams.eval_batch_size to account for the extra spin of the evaluator dataloaders
```
Comment (nik-mosaic) [inline tests/trainer/test_ddp.py:185]: Remove hparams. from comment

File: tests/trainer/test_ddp.py [hunk 3]
```diff
@@ -143,44 +105,67 @@ def test_ddp(device: str, world_size: int, dummy_model_hparams: ModelHparams, de
     and 2) each ddp process is indeed getting different data.
     """
 
-    dummy_model_hparams.num_classes = 100
-    model = dummy_model_hparams.initialize_object()
-    assert isinstance(model, SimpleModel)
+    model = SimpleModel(num_classes=100)
```
Comment (nik-mosaic) [inline tests/trainer/test_ddp.py:108]: Remove dummy_model_hparams from arguments of test_ddp() now that we're using SimpleModel

**Review by nik-mosaic — COMMENTED** [review]
Already much more readable---looking good. Just a couple of minor comments.

**Review by eracah — APPROVED** [review]
LGTM

**Pull request merged**